seep-master/src/main/java/uk/ac/imperial/lsds/seepmaster/query/QueryManager.java

```diff
@@ -233,7 +220,15 @@ private LogicalSeepQuery executeComposeFromQuery(String pathToJar, String defini
 		URLClassLoader ucl = new URLClassLoader(urls);
 		try {
 			baseI = ucl.loadClass(definitionClass);
-			baseInstance = baseI.newInstance();
+			// For backwards compatibility, use the default constructor if one with a string array argument is not found
+			try {
+				baseInstance = baseI.getConstructor(String[].class).newInstance((Object)queryArgs);
+			} catch (NoSuchMethodException e)  {
+				baseInstance = baseI.newInstance();
```

Comment:
Could potentially still fail if baseI has no default constructor or only a private one. Since you did not seem to care about this in the old version though, I think this should be good.